### PR TITLE
[feat] Add SITU trtllmgen MOE

### PR DESCRIPTION
**Pull request by jiahanc**
## 📌 Description

Add SiTU activation support for TRTLLM-Gen MoE for MXFP4 × MXFP8 and NVFP4 × NVFP4 

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **New Features**
  * Added SiTU gated activation support for FP4 MoE in routed and non-routed execution, with configurable per-expert `gemm1_alpha`, `gemm1_beta`, and `gemm1_clamp_limit`.
  * Extended FP4 MoE trace templates with an `activation_type` input to select SiTU.
* **Bug Fixes**
  * Improved activation-type mapping/validation so SiTU uses the correct gated behavior.
  * Made MoE kernel/tile selection dtype-aware (activation + weight) and corrected per-token scaling eligibility during config generation.
* **Tests**
  * Expanded MoE coverage for SiTU and optional per-expert parameters, including additional routed-logit and autotune regression checks.
* **Chores**
  * Updated the remote BMM artifact reference and checksum.

### CODE REVIEW

**Review by aleozlx — COMMENTED** [review]
code review: looks good

File: flashinfer/trace/templates/moe.py [hunk 1]
```diff
@@ -1170,7 +1171,28 @@ def _unpack_fp4_e2m1(packed: torch.Tensor) -> torch.Tensor:
...
+            -1.0,
+            -1.5,
+            -2.0,
+            -3.0,
+            -4.0,
+            -6.0,
+        ],
+        dtype=torch.float32,
+        device=packed.device,
+    )
```
Comment (rosenrodt) [inline flashinfer/trace/templates/moe.py]: Why replace _E2M1_LUT_VALUES with literal values?

Comment (jiahanc) [reply flashinfer/trace/templates/moe.py]: fixed

File: flashinfer/fused_moe/core.py [hunk 2]
```diff
@@ -3188,6 +3178,46 @@ def _validate_bf16_gemm1_activation_params(
...
 
+def _prepare_fp4_block_scale_gemm1_activation_params(
+    activation_type: int,
+    gemm1_alpha: Optional[torch.Tensor],
+    gemm1_beta: Optional[torch.Tensor],
+    gemm1_clamp_limit: Optional[torch.Tensor],
+    local_num_experts: int,
+    device: torch.device,
+) -> tuple[Optional[torch.Tensor], Optional[torch.Tensor], Optional[torch.Tensor]]:
+    if int(activation_type) != int(ActivationType.Situ):
```
Comment (IwakuraRein) [inline flashinfer/fused_moe/core.py]: Is this function really necessary? If flashinfer is running in eager mode, will it cause performance issues?

File: flashinfer/trace/templates/moe.py [hunk 3]
```diff
@@ -1292,6 +1298,29 @@ def _fp4_moe_run_experts(
     T = A.shape[0]
     output = torch.zeros((T, H), dtype=torch.float32, device=device)
     local_start = int(local_expert_offset)
+    activation_value = int(activation_type)
+    if activation_value not in (3, 10):
```
Comment (IwakuraRein) [inline flashinfer/trace/templates/moe.py]: Can you use `normalize_activation_type` to convert it to the enum, instead of int?


```suggestion
    activation_type = normalize_activation_type(activation_type)
    if activation_type not in (ActivationType.Swiglu, ActivationType.Situ):
```

File: flashinfer/trace/templates/moe.py [hunk 4]
```diff
@@ -1766,6 +1839,29 @@ def _moe_fp4_block_scale_init(
         on by default for parity with the test).
     Requires SM100+ at runtime; CPU smoke tests skip.
     """
+    situ_alpha = gemm1_alpha
+    situ_beta = gemm1_beta
+    situ_clamp = gemm1_clamp_limit
+    if int(activation_type) == 10:
```
Comment (IwakuraRein) [inline flashinfer/trace/templates/moe.py]: Same as above

**Review by IwakuraRein — APPROVED** [review]

**Review by samuellees — APPROVED** [review]
